### ISSUE TITLE
Fix Deprecated Buttons with `dart fix`

### ISSUE THREAD

**Issue by grafst** [dart-lang/sdk#51001]
# Problem:

I want to try out a package from pub.dev.

so I clone the repository, and run it:
```
gh repo clone guiguito/flutter_dollar_gesture_detector
cd flutter_dollar_gesture_detector
flutter run -d macos
```

It doesn't build because RaisedButton is now called ElevatedButton:
<img width="1125" alt="2023-01-12_18-09-31" src="https://user-images.githubusercontent.com/8471055/212133625-e619bf66-78e1-45af-85cc-6898706d61e1.png">

so I run `dart fix`. 
Which, by the way is such a brilliant idea. To provide me with a tool to automatically fix common problems.

...but it doesn't help:
<img width="535" alt="2023-01-12_18-12-47" src="https://user-images.githubusercontent.com/8471055/212134251-283ab877-9640-4c04-be80-ac41b8b3a0da.png">

now I have to start up my IDE and manually fix all those Issues 😢 
I just wanted to try out the package quickly....

* Dart SDK Version: 2.18.6
* OS: macOS 13.0.1

# Solution
`dart fix `automatically replaces the buttons according to this official documentation:
https://docs.flutter.dev/release/breaking-changes/buttons#restoring-the-original-button-visuals

-----

wouldn't that be cool? 😄

**Comment by bwilkerson** [comment]
@Piinks @goderbauer

**Comment by Piinks** [comment]
We investigated the above cases for snackbars and buttons, and found that we could not support the migrations with the dart fix tool. 
When these breaking changes were made, the dart fix tool did not exist. Since it has been added, we have been including it in the design process so that we can try to avoid making further changes that dart fix cannot handle.

For snackbars and buttons, migration guides are available on the website:
- https://docs.flutter.dev/release/breaking-changes/scaffold-messenger
- https://docs.flutter.dev/release/breaking-changes/buttons
